USAGE_EXTENSIBILITY.md

```diff
@@ -0,0 +1,136 @@
+## Extending Marked
+
+To champion the single-reponsibility and open/closed prinicples, we have tried to make it relatively painless to extend marked. If you are looking to add custom functionality, this is the place to start.
+
+<ul>
+	<li><a href="#renderer">The renderer</a></li>
+	<li><a href="#lexer">The lexer</a></li>
+	<li><a href="#parser">The parser</a></li>
+</ul>
+
+<h2 id="rederer">The renderer</h2>
```

Comment:
The renderer...

Reply:
Why are you using HTML here? Instead of `<h2>` you can use `## Renderer`

Reply:
Fair question. For the `id` attribute. Is there another way to get inline anchors to work properly using GitHub Markdown??

Reply:
Consolidating comments from #1061: I've always run into issues there [GitHub automatically generating IDs]; so, normally end up doing it by hand. Also, could make it more difficult to use in a non-GFM setting; at least here we know the TOC will always work[...like pulling in the Markdown file into a separate context outside of the GFM markdown rendering.]

Reply:
I typically let github generate it like so: https://github.com/styfle/copee#demo
But a TOC is a good point, you don't want to accidentally break it if you change the heading later.